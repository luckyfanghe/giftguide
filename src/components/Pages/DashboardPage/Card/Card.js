import React, { Component } from "react";
import { Button } from "react-bootstrap";
import * as S from "./styled";

const Card = props => {
  // const classes = useStyles();
  let description;
  switch (props.title) {
    case "CALENDAR":
      description = [
        <S.DescriptionContent id="des-1" key="des-1">
          gift guide creates a customized
        </S.DescriptionContent>,
        <S.DescriptionContent id="des-2" key="des-2">
          calendar for you with all your loved
        </S.DescriptionContent>,
        <S.DescriptionContent id="des-3" key="des-3">
          one's most important dates, in one
        </S.DescriptionContent>,
        <S.DescriptionContent id="des-4" key="des-4">
          easy location
        </S.DescriptionContent>
      ];
      break;
    case "WISH LIST":
      description = [
        <S.DescriptionContent id="des-5" key="des-5">
          gift guide users set up profiles
        </S.DescriptionContent>,
        <S.DescriptionContent id="des-6" key="des-6">
          containing their favorite stores, wish
        </S.DescriptionContent>,
        <S.DescriptionContent id="des-7" key="des-7">
          lists and items they love, so you no
        </S.DescriptionContent>,
        <S.DescriptionContent id="des-8" key="des-8">
          longer have to ask for ideas or sizes
        </S.DescriptionContent>
      ];
      break;
    case "REMINDERS":
      description = [
        <S.DescriptionContent id="des-9" key="des-9">
          gift guide sends you reminders before
        </S.DescriptionContent>,
        <S.DescriptionContent id="des-10" key="des-10">
          your loved ones' events with their
        </S.DescriptionContent>,
        <S.DescriptionContent id="des-11" key="des-11">
          wish lists and favorite stores so you
        </S.DescriptionContent>,
        <S.DescriptionContent id="des-12" key="des-12">
          never miss an important date again
        </S.DescriptionContent>
      ];
      break;
    default:
      break;
  }

  return (
    <React.Fragment>
      <S.Body id={props.id} key={props.id} data-testid="Card">
        <S.Title>{props.title}</S.Title>
        <S.DescriptionContainer>{description}</S.DescriptionContainer>
        <S.Button> {props.buttonTitle} </S.Button>
      </S.Body>
    </React.Fragment>
  );
};

export default Card;
