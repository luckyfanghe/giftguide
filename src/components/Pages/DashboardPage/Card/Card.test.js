import React from "react";
import { render, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import Card from "./Card";

it("Card renders corrently", () => {
  const { queryByTestId } = render(<Card />);

  expect(queryByTestId("Card")).toBeTruthy();
});

describe("Card", () => {
  test("Card snapshot renders ", () => {
    const component = renderer.create(<Card />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
