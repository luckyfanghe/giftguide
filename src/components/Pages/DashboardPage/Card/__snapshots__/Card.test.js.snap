// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Card snapshot renders  1`] = `
<div
  className="sc-AykKC eZouOZ"
  data-testid="Card"
>
  <div
    className="sc-AykKD jerNDl"
  />
  <div
    className="sc-AykKF hnEzuj"
  />
  <button
    className="sc-AykKE dosAzx"
    type="button"
  >
     
     
  </button>
</div>
`;
